Docs/source/Software/BSP.rst

```diff
@@ -0,0 +1,51 @@
+***********************
+Board Support Package
+***********************
+
+I2C BSP: Manthan Upadhyaya
+=================================
+
+Purpose
+    The I2C driver is interrupt driven and is used to communicate with the EEPROM to log data. It 
+    requires pins PA8(SCL) and PC9(SDA) for the I2C3.
+
+Usage
+    There are multiple definitions of the same functions within this library. They are split into 2 
+    categories, one for the RTOS and one for the Bare-Metal version of the BPS. There are ``#define`` that 
+    encompass these functions that are set when compiled. The default ``#define`` is "RTOS" but it can be 
+    changed to compile the Bare-Metal code if necessary. The ``BSP_I2C.h`` file has more information on how
+    to use each function. The same function declarations are used regardless of whether it is RTOS or
+    Bare-Metal.
+
+Additional Considerations
+    None
+
+UART BSP: Manthan Upadhyaya
+==================================
+
+Purpose
+    The UART driver is interrupt driven and initializes 2 serial transmission lines(UART 2 and 3). 
+    USART 2 is for the Bluetooth module and uses pins PA2 and PA3. USART3 is for the USB and uses pins
+    PC5 and PB10.
+
+Usage
+    This driver is also split into two sections just like the I2C driver. One is for RTOS and the 
+    other is for Bare-Metal. The ``BSP_UART.h file`` has more information on how to use each function.
+
+Additional Considerations
+    Since the BPS cannot be changed during the race, the BLE module must be disabled before the race
```

Comment:
Can you put the regulation number for why the BPS firmware and settings cannot be changed during the race?